src/api/route-services/access.js

```diff
@@ -0,0 +1,104 @@
+
+const _ = require('lodash');
+
+const config = require('../../config');
+const AWS = require('../../utils/requireAWS');
+const getLogger = require('../../utils/getLogger');
+
+const {
+  createDynamoDbInstance,
+  convertToJsObject,
+  convertToDynamoDbRecord,
+} = require('../../utils/dynamoDb');
+const { isRoleAuthorized } = require('./roles');
+
+
+const logger = getLogger('[AccessService] - ');
+
+class AccessService {
+  constructor() {
+    this.userAccessTableName = `user-access-${config.clusterEnv}`;
+    this.inviteAccessTableName = `invite-access-${config.clusterEnv}`;
+  }
+
+  // grantRole adds or replaces role for userId and experimentId
+  async grantRole(userId, experimentId, projectId, role) {
+    logger.log(`PUT experiment role ${experimentId} data`);
+
+    const params = {
+      TableName: this.userAccessTableName,
+      Item: {
+        userId,
+        experimentId,
+        projectId,
+        role,
+      },
+    };
+
+    const docClient = new AWS.DynamoDB.DocumentClient({
+      region: config.awsRegion,
+    });
+
+    await docClient.put(params).promise();
+  }
+
+  async canAccessExperiment(userId, experimentId, url, method) {
+    logger.log(`GET experiment permissions ${experimentId} data`);
+    const key = convertToDynamoDbRecord({
+      experimentId,
+      userId,
+    });
+
+    const params = {
+      TableName: this.userAccessTableName,
+      Key: key,
+    };
+
+    console.log(params);
+
+    const dynamodb = createDynamoDbInstance();
```

Comment:
we don't need to create a new instance of dynamodb for every particular class function -- we can make `dynamodb` a class variable

Reply:
I agree and that is what I tried to do. However, the way we mock AWS then does not work (because the instance is created when we import it). I considered whether refactoring this but I thought it should not be part of this PR because all of the route-services do this (creating an instance every time).